### DISCUSSION TITLE
JSON-WSP may need to be adjusted to nominate `BootstrapWitness` fields as required

### DISCUSSION THREAD

**Discussion by ivaylo-andonov** [CardanoSolutions/ogmios#285] [Archives]
**Describe your idea, in simple words.**

Currently, Ogmios schema exposes all [BootstrapWitness](https://github.com/CardanoSolutions/ogmios/blob/master/clients/TypeScript/packages/schema/src/index.ts#L1294-L1299) fields as `optional` and two of them `nullable`, however in [CDDL](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/babbage.cddl#L356-L361) they seem to be `required`.

**What should be the expected behavior?**

There is no type discrepancy between Ogmios schema definition and CDDL.

**Comment by KtorZ** [comment]
Hey, thanks for the catch!

- The fields being optional is definitely wrong. That's a mistake in the JSON-schema (a missing `required`).
- The `addressAttributes` can be null because only very old Byron addresses had such attributes; the new kind of bootstrap addresses that came after (so-called "Icarus Addresses") do not; while on-chain this is still serialized as an empty byte string, the choice at the time was to simply mark it as `null` instead of returning an empty byte string. 

Having said that, I can see how the latter is confusing; I'll add it to the wishlist of [Ogmios V6 breaking changes](https://github.com/CardanoSolutions/ogmios/discussions/263) which are coming soon.

**Reply by ivaylo-andonov** [reply]
Thanks for the fast response! @KtorZ
